deployment/terraform/cloudbuild-tf-plan.yaml

```diff
@@ -0,0 +1,28 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+# run terraform plan for oss-vdb-test
+# TODO: regenerate the api protobufs?
+- name: gcr.io/oss-vdb/terraform
+  args: ['init', '-no-color']
+  dir:  deployment/terraform/environments/oss-vdb-test
+- name: gcr.io/oss-vdb/terraform
+  args: ['plan', '-no-color']
+  dir:  deployment/terraform/environments/oss-vdb-test
+# TODO(michaelkedar): plan for oss-vdb
+
+# Set in Cloud Build:
+# serviceAccount: 'projects/oss-vdb/serviceAccounts/deployment@oss-vdb.iam.gserviceaccount.com'
```

Comment:
Just calling this out:
This service account has a lot of permissions, and Terraform lets you execute arbitrary scripts in `plan` (and `apply`).

You do need to do the /gcbrun comment for external contributors, though maybe the planner should have its own service account with the minimum required permissions to be safe.

Reply:
Do you know what the minimum required permissions are?

Reply:
From some playing around, this seems to work for the planner:

Roles on the project:
- Viewer
- Pub/Sub Admin

On the Terraform bucket
- Storage Legacy Bucket Owner
- Storage Legacy Object Reader

It might be possible to make a more minimum set with a custom role, but I'd need a deeper look into it.

Reply:
That seems like a pretty limited set to begin with already. Let's create a new one for this. 

Is the Pub/Sub one definitely needed?

Reply:
Is deployment@oss-vdb.iam.gserviceaccount.com a custom-created service account for this? https://cloud.google.com/deploy/docs/cloud-deploy-service-account#execution_service_account is saying to me that the default compute account is normally used. We should definitely follow the principle of least privilege here.

Reply:
Yeah, Pub/Sub Admin is a strange one, but Pub/Sub Editor and Pub/Sub Viewer don't seem to give permissions to look at publishers and subscribers for some reason.

Reply:
> Is [deployment@oss-vdb.iam.gserviceaccount.com](mailto:deployment@oss-vdb.iam.gserviceaccount.com) a custom-created service account for this? https://cloud.google.com/deploy/docs/cloud-deploy-service-account#execution_service_account is saying to me that the default compute account is normally used. We should definitely follow the principle of least privilege here.

The deployment service account already existed (for App Engine iirc) I was just reusing it. Will make a dedicated "terraform plan" service account for this one

Reply:
Created a terraform-plan@oss-vdb.iam.gserviceaccount.com for the planner.